allennlp_models/common/model_card.py

```diff
@@ -0,0 +1,267 @@
+
+    def to_dict(self):
+        info = {}
+        for key, val in self.__dict__.items():
+            if val:
+                info["evaluation_" + key] = val
+        return info
+
+
+@dataclass
+class TrainingData(ModelCardInfo):
+    """
+    This provides information about the training data.
+    """
+
+    dataset: Optional[str] = None
+    motivation: Optional[str] = None
+    preprocessing: Optional[str] = None
+
+    def to_dict(self):
+        info = {}
+        for key, val in self.__dict__.items():
+            if val:
+                info["training_" + key] = val
+        return info
+
+
+@dataclass
+class QuantitativeAnalyses(ModelCardInfo):
+    """
+    This provides disaggregated evaluation of how the
+    model performed based on chosen metrics.
+    """
+
+    unitary_results: Optional[str] = None
+    intersectional_results: Optional[str] = None
+
+
+@dataclass
+class EthicalConsiderations(ModelCardInfo):
+    """
+    This highlights any ethical considerations to keep
+    in mind when using the model.
+    Eg. Is the model intended to be used for informing
+    decisions on human life? Does it use sensitive data?
+    What kind of risks are possible, and what mitigation
+    strategies were used to address them?
+    """
+
+    ethical_considerations: Optional[str] = None
+
+
+@dataclass
+class CaveatsAndRecommendations(ModelCardInfo):
+    """
+    This lists any additional concerns. Eg. were any relevant
+    groups not present in the evaluation data?
+    """
+
+    caveats_and_recommendations: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class ModelCard(ModelCardInfo):
+    """
+    The model card stores the recommended attributes for model reporting
+    as described in the paper [Model Cards for Model Reporting (Mitchell et al, 2019)]
+    (https://arxiv.org/pdf/1810.03993.pdf).
+    """
+
+    id: str
+    name: Optional[str] = None
+    archive_file: Optional[str] = None
+    overrides: Optional[Mapping] = None
+    model_details: Optional[ModelDetails] = None
+    intended_use: Optional[IntendedUse] = None
+    factors: Optional[Factors] = None
+    metrics: Optional[Metrics] = None
+    evaluation_data: Optional[EvaluationData] = None
+    training_data: Optional[TrainingData] = None
+    quantitative_analyses: Optional[QuantitativeAnalyses] = None
+    ethical_considerations: Optional[EthicalConsiderations] = None
+    caveats_and_recommendations: Optional[CaveatsAndRecommendations] = None
+
+    def to_dict(self):
+        info = {}
+        for key, val in self.__dict__.items():
+            if key != "id":
+                if isinstance(val, ModelCardInfo):
+                    info.update(val.to_dict())
+                else:
+                    info[key] = val
+        return info
+
+
+def add_pretrained_model(
+    id: str,
+    model_class: Optional[type] = None,
+    name: Optional[str] = None,
+    archive_file: Optional[str] = None,
+    overrides: Optional[Mapping] = None,
```

Comment:
Overrides are supposed to override the model's architecture defaults. I included them because of models like this one:

```{
    "id": "nmn",
    "archive_file": "https://storage.googleapis.com/allennlp-public-models/drop-nmn-2020.04.04.tar.gz",
    "predictor_name": "drop_demo_predictor",
    "overrides": {
        "model": {
            "debug": true,
            "beam_size": 1
       }
    }
}```

Reply:
We apply those values to the config after it is loaded from the archive? I didn't know we had that capability, but I immediately know what to use it for :-)